exercises/practice/book-store/.meta/example.rb

```diff
@@ -1,46 +1,29 @@
+require 'set'
+
 class BookStore
 
-  GROUP_DISCOUNTS = [0, 0.05, 0.1, 0.2, 0.25]
+  GROUP_DISCOUNTS = [1.0, 0.95, 0.9, 0.8, 0.75]
   INDIVIDUAL_PRICE = 8
 
   def self.calculate_price(basket)
-    #FIXME: find a real solution instead of hardcoding :(
-    return 51.20 if basket == [1, 1, 2, 3, 4, 4, 5, 5]
+    return 0.00 if basket.empty?
 
     groups = []
-    remaining_books = basket.dup
-
-    # Make as many groups of 4 as you can
-    while (largest_group = remaining_books.uniq).length > 3
-      group_of_four = largest_group[0..3]
-      group_of_four.each {|book| remaining_books.delete_at(remaining_books.index(book)) }
-      groups << group_of_four
-    end
-
-    # Bump as many of them to groups of 5 as you can
-    groups.each do |group|
-      fifth_book = (remaining_books - group).first
-      next unless fifth_book
-
-      group << fifth_book
-      remaining_books.delete_at(remaining_books.index(fifth_book))
-    end
 
     # Make the largest groups you can with the remaining books
-    while (new_group = remaining_books.uniq).any?
-      new_group.each {|book| remaining_books.delete_at(remaining_books.index(book)) }
-      groups << new_group
+    while !basket.empty?
```

Comment:
The new loop tests the basket that the books come out of, whereas the old loop tests the group that the books go in to.  Sort of a 'Six of one, half a dozen of the other' situation.  And yes, I should have used 'until' instead of negating 'while', but it doesn't matter now, as I'm considering just deleting the loop entirely and sticking with the old one (albeit with one slight change)